R/bm-write-file.R

```diff
@@ -25,7 +25,7 @@ write_file <- Benchmark("file-write",
       if (file_type == "feather") {
         return(function(...) arrow::write_feather(..., compression = compression))
       } else if (file_type == "parquet") {
-        return(function(...) arrow::write_parquet(..., compression = compression))
+        return(function(...) arrow::write_parquet(make_table_parquet_compatible(...), compression = compression))
```

Comment:
I don't think we should add `make_table_parquet_compatible` here. There shouldn't be a huge amount of overhead, but we don't want to add to our writing benchmark with additional type checking (and casting). IIUC, none of our write benchmarks are failing, but if they start to due to this, we should update the data we're using as an input rather than cast during the benchmark.